src/RAM/Science/components/rocklookup.tsx

```diff
@@ -0,0 +1,706 @@
+        this.setState({ s_Texture, availTextures }, () => this.reloadOptions())
+        break
+    }
+  }
+
+  clearSelectedFeatures(): void {
+    this.setState(
+      {
+        s_Colors: [],
+        s_Cleave: [],
+        s_Forms: [],
+        s_Texture: [],
+      },
+      () => this.reloadOptions()
+    )
+  }
+
+  possibleRocks(): JSX.Element | void {
+    if (this.state.outputArr.length > 0) {
+      return (
+        <div style={{ ...row }}>
+          <div style={{ ...row, flexGrow: 1 }}>
+            <div style={{ ...column, flexGrow: 1, backgroundColor: "#ddd", width: "40%" }}>
+              <p style={{ fontWeight: "bold", textAlign: "center" }}>
+                {this.state.outputArr[this.state.selectedOutput].Rock.name}
+              </p>
+              <p style={{ textAlign: "center", justifyContent: "center" }}>
+                {this.state.outputArr[this.state.selectedOutput].Rock.description}
+              </p>
+            </div>
+            <div style={{ width: "calc(60% - 62px)", alignSelf: "center" }}>
+              <img
+                style={{ width: "100%" }}
+                src={path.join(
+                  __dirname,
+                  `../assets/rockLookupAssets/images/${this.state.outputArr[this.state.selectedOutput].Rock.name}.png`
+                )}
+                alt={this.state.outputArr[this.state.selectedOutput].Rock.name}
+              ></img>
+            </div>
+          </div>
+          <div style={{ ...column, alignSelf: "right" }}>
+            <button
+              style={{ width: "62px", flexGrow: 1 }}
+              type={"button"}
+              onClick={() =>
+                this.setState({
+                  selectedOutput: Math.max(0, this.state.selectedOutput - 1),
+                })
+              }
+            >
+              ^^^
+            </button>
+            <p style={{ margin: "0 5 0 5", textAlign: "center" }}>
+              {this.state.selectedOutput + 1} / {this.state.outputArr.length}
+            </p>
+            <button
+              style={{ width: "62px", flexGrow: 1 }}
+              type={"button"}
+              onClick={() =>
+                this.setState({
+                  selectedOutput: Math.min(this.state.selectedOutput + 1, this.state.outputArr.length - 1),
+                })
+              }
+            >
+              vvv
+            </button>
+          </div>
+        </div>
+      )
+    }
+  }
+
+  render(): JSX.Element {
+    return (
+      <div style={this.props.style}>
+        <div style={label}>Rock Lookup</div>
+        <div style={container}>
+          {/*<div style={row}>
+            <form onSubmit={e => this.handleFeatureSubmit(e)}>
+              <input
+                type="text"
+                name="newTag"
+                onChange={e => {
+                  this.handleInputChange(e)
+                }}
+              />
+              <input type="submit" value={this.state.searchField} />
```

Comment:
Is the search bar still planned?

Reply:
For now yes. It will largely depend on Mercedes and a number of different factors. If the size of the database grows to a large enough volume, it might actually be faster to type known feature descriptors rather than search for them in the menus